bindings/python/examples/iio_writedev.py

```diff
@@ -0,0 +1,221 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+import time
+
+import iio
+import getopt
+import sys
+import signal
+
+
+class Option:
+    def __init__(self, name, alias, description):
+        self.name = name
+        self.alias = alias
+        self.description = description
+
+
+options = [
+    Option('help', 'h', 'Show this help and quit.'),
+    Option('network', 'n', 'Use the network backend with the provided hostname.'),
+    Option('uri', 'u', 'Use the context with the provided URI.'),
+    Option('buffer-size', 'b', 'Size of the capture buffer. Default is 256.'),
+    Option('samples', 's', 'Number of samples to capture, 0 = infinite. Default is 0.'),
+    Option('timeout', 'T', 'Buffer timeout in milliseconds. 0 = no timeout'),
+    Option('auto', 'a', 'Scan for available contexts and if only one is available use it.'),
+    Option('cyclic', 'c', 'Use cyclic buffer mode.')
+]
+
+usage_message = 'Usage:\n\t' \
+                'iio_readdev [-n <hostname>] [-T <timeout-ms>] [-b <buffer-size>] ' \
+                '[-s <samples>] [-c <cyclic>] <iio_device> [<channel> ...]\n'
+
+unixOptions = 'hn:u:b:s:T:ac'
+gnuOptions = ['help', 'network', 'uri', 'buffer-size', 'samples', 'timeout', 'auto', 'cyclic']
+
+
+def get_device_and_channels(arguments):
+    arguments = iter(arguments)
+    arg = next(arguments, None)
+
+    while True:
+        if arg == '-h' or arg == '--help' or arg == '-a' or arg == '--auto' or arg == '-c' or arg == '--cyclic':
+            arg = next(arguments, None)
+        elif arg == '-n' or arg == '--network' or arg == '-u' or arg == '--uri' or \
+                arg == '-b' or arg == '--buffer-size' or arg == '-s' or arg == '--samples' or \
+                arg == '-T' or arg == '--timeout':
+            next(arguments)
+            arg = next(arguments, None)
+        else:
+            break
+
+    # arg should contain the device name or None if there is no device
+    ret_list = [arg]
+
+    while arg is not None:
+        arg = next(arguments, None)
+
+        if arg is not None:
+            ret_list += [arg]
+
+    return ret_list
+
+
+def usage():
+    print(usage_message)
+    print('Options:')
+    for option in options:
+        print('\t-' + option.alias + ', --' + option.name)
+        print('\t\t\t', option.description)
+
+
+def keyboard_interrupt_handler(signal, frame):
+    sys.exit(0)
+
+
+signal.signal(signal.SIGINT, keyboard_interrupt_handler)
+
+
+def main():
+    raw_arguments = sys.argv[1:]
+
+    try:
+        arguments, values = getopt.getopt(raw_arguments, unixOptions, gnuOptions)
```

Comment:
Why was argparse not used? It much much simpler and in the core library.

Reply:
I didn't know argparse was in the core library; getopt just seemed simpler at the first glance for me. I have changed them to use argparse.